lib/dalli/pipelined_getter.rb

```diff
@@ -92,29 +50,101 @@ def abort_without_timeout(servers)
       servers.each(&:pipeline_abort)
     end
 
-    def fetch_responses(servers, start_time, timeout, &block)
+    def fetch_responses(requests, timeout, &block)
+      # FIXME: this was here. why. where should it go?
       # Remove any servers which are not connected
-      servers.delete_if { |s| !s.connected? }
-      return [] if servers.empty?
+      # servers.delete_if { |s| !s.connected? }
+
+      start_time = Time.now
+      servers = requests.keys
+
+      # FIXME: this was executed before the finish request was sent. Why?
```

Comment:
I have to look at this in a little detail.  The behavior in the case of no servers available is supposed to a returned `[]` with no error.  But if I recall correctly, there are cases where the a server is not connected that can trigger raising a `Dalli::Error` without this check.  I'll find some time to look.

Reply:
Any hints on the intended original behavior of the multi-server code are highly appreciated. I haven't been able to fully wrap my head around it yet, and I guess it is the hardest to get right when things fail in between reading/writing chunks. Hard to correctly test as well..